src/transports/ipc.rs

```diff
@@ -183,30 +212,29 @@ async fn run_server(unix_stream: UnixStream, messages_rx: mpsc::UnboundedReceive
                 Some(Ok(bytes)) => {
                     read_buffer.extend_from_slice(&bytes);
 
-                    let pos = match read_buffer.iter().rposition(|&b| b == b']' || b == b'}') {
-                        Some(pos) => pos + 1,
-                        None => continue,
-                    };
+                    let read_len = {
+                        let mut de: serde_json::StreamDeserializer<_, serde_json::Value> =
```

Comment:
That's awesome! I hope it works in case there is no `\n` between requests, right? I remember this being the biggest issue, but reading the documentation it seems it does so I'm super happy to get rid of the hacky code we had previously.